tekore/client/process.py

```diff
@@ -27,3 +33,18 @@ def model_list(type_: type, top_item: str):
     def post_func(json: dict):
         return ModelList(type_(**i) if i is not None else None for i in json[top_item])
     return post_func
+
+
+paging_type = {
+    'artist': FullArtistOffsetPaging,
+    'album': SimpleAlbumPaging,
+    'playlist': SimplePlaylistPaging,
+    'track': FullTrackPaging,
+}
+
+
+def search_result(json: dict):
+    """
+    Unpack search result dicts into respective paging type constructors.
+    """
+    return tuple(paging_type[key[:-1]](**json[key]) for key in json.keys())
```

Comment:
One thought here: maybe this code should go to `api/search.py`, since it is too specific and used only for processing search result.

Reply:
Most definitely. I did a similar thing for paging 👍